Adhere to Mac platform when showing shortcuts on a Mac

jasmussen opened the pull request:
This fixes #9226.

This changes the display of keyboard shortcuts on the Mac to:

- not have the plus between
- use unicode symbols for the characters

The larger discussion for why we're doing this is in the ticket referenced, but the gist of it is: this is what MacOS does, and we mean to tap into that same pattern that Mac users will be used to.

Screenshots:

![screen shot 2018-08-29 at 14 19 46](https://user-images.githubusercontent.com/1204802/44787559-cba80e00-ab97-11e8-883f-ef3362dc0fa7.png)

![screen shot 2018-08-29 at 14 24 45](https://user-images.githubusercontent.com/1204802/44787562-ccd93b00-ab97-11e8-8d14-2441ccfc8df2.png)

![screen shot 2018-08-29 at 14 24 54](https://user-images.githubusercontent.com/1204802/44787564-ce0a6800-ab97-11e8-8e6a-968df44646f7.png)

![screen shot 2018-08-29 at 14 24 59](https://user-images.githubusercontent.com/1204802/44787569-cfd42b80-ab97-11e8-820c-133bb5fd5417.png)

![screen shot 2018-08-29 at 14 28 49](https://user-images.githubusercontent.com/1204802/44787598-df537480-ab97-11e8-9f8e-7633187a80c1.png)

jasmussen committed:
Adhere to Mac platform when showing shortcuts on a Mac

This fixes #9226.

This changes the display of keyboard shortcuts on the Mac to:

- not have the plus between
- use unicode symbols for the characters

The larger discussion for why we're doing this is in the ticket referenced, but the gist of it is: this is what MacOS does, and we mean to tap into that same pattern that Mac users will be used to.

jasmussen commented:
Note that I also changed "bksp" to "backspace" for the backspace keyboard shortcut. But I think I saw PRs fly by that changes this keyboard combo entirely anyway, so perhaps not worth spending too much time worrying about that one.

jasmussen commented:
Totally unrelated to this PR, but is there _anything at all_ we can do to not set focus immediately on the close button when a modal opens? I'd prefer nearly any other option 😅

ZebulanStanphill commented:
Backspace has a symbol used for it in macOS: "⌫".

https://support.apple.com/guide/mac-help/what-are-those-symbols-shown-in-menus-cpmh0011/mac

jasmussen commented:
That's interesting Zeb, thanks for sharing the link. If we keep the Backspace option, would be worth looking into ambiguating that too. But this is the shortcut that shuts down Gnome, right?

ZebulanStanphill commented:
Actually, I use KDE Plasma, not GNOME. But the desktop environment is irrelevant. The shortcut actually shuts down (or in some cases reboots) the X display server, which is currently the most common display server used in Linux distros. Some distros (like Ubuntu) disable the shortcut, but not Arch Linux or Antergos (the Arch-based distro I use), apparently.

tofumatt commented on packages/editor/src/components/block-settings-menu/index.js:
```diff
@@ -38,7 +38,7 @@ const shortcuts = {
 	removeBlock: {
 		raw: rawShortcut.primaryAlt( 'backspace' ),
-		display: displayShortcut.primaryAlt( 'bksp' ),
+		display: displayShortcut.primaryAlt( 'Backspace' ),
```
Might as well change this to the Mac one too, though I guess you'd need to add a new keyboard constant for it.

jasmussen replied:
Well I was going to, but then I discovered the `isMac` variable is not present in this component, and so instead of finding out how to make that happen, I had to move on to other work.

Given the fact that this shortcut will change in the very near future (it was already changed once and that change was reverted), I feel like it might not be worth it to change this when it needs to be something else? See discussion in https://github.com/WordPress/gutenberg/pull/9190.

tofumatt reviewed (approved):
I say we should add the Backspace shortcut too, but as-is that's cool, go for it 😄

youknowriad commented:
Noting that the unit tests are failing here, probably just a small breakage.

jasmussen commented:
@youknowriad I'm seeing this test failing:

<img width="733" alt="screen shot 2018-08-30 at 10 59 06" src="https://user-images.githubusercontent.com/1204802/44841492-c30d1180-ac43-11e8-93be-183e3934e967.png">

But I'm unsure how my code could've affected that.

youknowriad commented:
This was is intermittent and is passing on Travis, I was talking about unit test `npm run test-unit`

youknowriad commented:
I can take a look a bit after :)

jasmussen commented:
Looking, thanks.

Commit:
Fix tests.

jasmussen commented:
I believe I fixed the tests now.

Note that I was a little bit confuse as to whether I fixed the test _titles_ correctly, feel free to have a look.

youknowriad commented:
Looks good to me, let's wait for travis :)

jasmussen commented:
All green, Riad. Good to merge?

youknowriad commented:
Merge? Merge what?
